src/module_wrap.cc

```diff
@@ -586,9 +592,9 @@ Maybe<URL> ResolveMain(Environment* env, const URL& search) {
     return Nothing<URL>();
   }
   if (!ShouldBeTreatedAsRelativeOrAbsolutePath(pjson.main)) {
-    return Resolve(env, "./" + pjson.main, search, IgnoreMain);
+    return ResolveRecurse(env, "./" + pjson.main, search, IgnoreMain);
```

Comment:
This can probably be simplified to a `ResolveRelativeWithExtensions`, that doesn't need to do the absolute checks and assumes its input is a relative URL.